frontend/javascripts/oxalis/model/bucket_data_handling/pullqueue.js

```diff
@@ -84,8 +94,19 @@ class PullQueue {
 
     const { renderMissingDataBlack } = Store.getState().datasetConfiguration;
 
+    const { signal } = this.abortController;
     try {
-      const bucketBuffers = await requestWithFallback(layerInfo, batch);
+      const bucketBuffers = await new Promise(async (resolve, reject) => {
+        const abort = () => reject(PULL_ABORTION_ERROR);
+        signal.addEventListener("abort", abort);
+        try {
```

Comment:
This try/catch inside a promise inside a try/catch construction is sub-optimal to say the least, but it's the only way I got it to work. The inner try/catch is needed to avoid unhandled promise rejections. The promise is needed to be able to trigger an error from the event listener that is catched by the outer try/catch. Please let me know if you see a better solution :see_no_evil:

Reply:
Since there is an inner and outer promise, it makes sense to me that there has to be an inner and outer try-catch. So, I'm assuming this cannot easily be avoided. However, maybe the abortable-logic can be extracted into a helper? A usage like this might be more concise (and reusable):

```js
try {
  const bucketBuffers = await asAbortable(
    requestWithFallback(layerInfo, batch),
    this.abortController.signal,
    PULL_ABORTION_ERROR
  );
  ...
} catch (...) {
  ...
}
```

with `asAbortable` looking somewhat like this:

```js
function asAbortable(promise, signal, error) {
  return new Promise(async (resolve, reject) => {
        const abort = () => reject(error);
        signal.addEventListener("abort", abort);
        try {
          const value = await promise;
          resolve(value);
        } catch (error) {
          reject(error)
        }
        signal.removeEventListener("abort", abort);
}
```